x/ethbridge/types/msgs.go

```diff
@@ -138,9 +139,9 @@ func (msg MsgBurn) ValidateBasic() error {
 		return ErrInvalidBurnSymbol
 	}
 
-	// check that enough ceth is sent to cover the gas cost.
-	if msg.CethAmount.LT(sdk.NewInt(burnGasCost)) {
-		return ErrCethAmount
+	// check that enough native_token is sent to cover the gas cost.
+	if msg.NativeTokenAmount.LT(sdk.NewInt(burnGasCost)) {
```

Comment:
This is incorrect.

We should query the keeper and check that the gas cost for this network is sufficient before continuing here.

Reply:
Each network should have an associated gas cost.

Reply:
Map networkid => {gasCost and currency to pay unpeg fee}

We should also assert that the msg.NativeTokenSymbol is the correct symbol.

Reply:
will add gas cost for each network.

Reply:
the gasCost will be checked when process the message. The validation here in msg.go should just depend on some static data. there is no approach to get the keeper or other context here.